### PR TITLE
GitSync Diff Only Option

### PR DESCRIPTION
**Pull request by NOYB**
--diff
 SSD Friendlier
 Same as default but only copies files that are different, instead of every thing even though identical.

### CODE REVIEW

**Comment by rbgarga** [comment]
I like the idea but I don't think it's necessary to create a .inc file that will be only used by one consumer. Could you please revert this part of the change?

**Comment by NOYB** [comment]
How can the function redefinition that prevents running gitsync again in the same dev shell session (command recall) be resolved?  That's why I moved them to an include file and used require once.
Is there a better way?  Thanks.

File: src/etc/phpshellsessions/gitsync [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*   cvs_sync
+<?/*   cvs_sync
```
Comment (NewEraCracker) [inline src/etc/phpshellsessions/gitsync]: Is this change intended?

Comment (NOYB) [inline src/etc/phpshellsessions/gitsync]: No.

**Comment by NewEraCracker** [comment]
I think this would be a good addition for 2.3.2.
Please include it. Thanks.

**Comment by cbuechler** [comment]
@NOYB could you please re-submit as a single commit, it's cleaner and easier to cherry-pick that way. The end result here looks fine, I can get that merged promptly.
